File: JAVA_ADVANCE/3Layers/src/main/java/com/vti/entity/Account.java
package com.vti.entity;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.Formula;


@Entity
@Table(name = "`Account`", catalog = "TestingSystem")
@Inheritance(strategy = InheritanceType.JOINED)
public class Account implements Serializable {

	private static final long serialVersionUID = 1L;

	@Column(name = "AccountID")
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private short id;

	@Column(name = "Email", nullable = false, length = 50, unique = true, updatable = true)
	private String email;

	@Column(name = "Username", nullable = false, length = 50, unique = true, updatable = true)
	private String userName;

	@Column(name = "FirstName", nullable = false, length = 50)
	private String firstName;

	@Column(name = "LastName", nullable = false, length = 50)
	private String lastName;

	@Formula("concat(FirstName,' ',LastName)")
	private String fullName;

	@Column(name = "CreateDate", updatable = true)
	@Temporal(TemporalType.TIMESTAMP)
	@CreationTimestamp
	private Date createDate;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "DepartmentID", nullable = false, referencedColumnName = "DepartmentID")
	private Department department;

//	@ManyToOne
//	@JoinColumn(name = "PositionID", nullable = false, referencedColumnName = "PositionID")
//	private Position position;

//	@ManyToOne
//	@JoinColumn(name = "SalaryID", nullable = false, referencedColumnName = "SalaryID")
//	private Salary salary;
//
//	@OneToOne(mappedBy = "creator")
//	private Group group;
//
//	@OneToMany(mappedBy = "account")
//	List<GroupAccount> groups;

	public Account() {

	}

	public Account(short id, String email, String userName) {
		this.id = id;
		this.email = email;
		this.userName = userName;	
	}
	public Account(String email, String userName, String firstName, String lastName) {
		this.email = email;
		this.userName = userName;
		this.firstName = firstName;
		this.lastName = lastName;
	}

//	public Position getPosition() {
//		return position;
//	}
//
//	public void setPosition(Position position) {
//		this.position = position;
//	}
//
//	public Salary getSalary() {
//		return salary;
//	}
//
//	public void setSalary(Salary salary) {
//		this.salary = salary;
//	}

	public short getId() {
		return id;
	}

	public void setId(short id) {
		this.id = id;
	}

	public String getEmail() {
		return email;
	}

	public String getFullName() {
		return fullName;
	}

	public void setFullName(String fullName) {
		this.fullName = fullName;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getUserName() {
		return userName;
	}

	public void setUserName(String userName) {
		this.userName = userName;
	}

	public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName;
	}

	public Date getCreateDate() {
		return createDate;
	}

	public void setCreateDate(Date createDate) {
		this.createDate = createDate;
	}

	public Department getDepartment() {
		return department;
	}

	public void setDepartment(Department department) {
		this.department = department;
	}

	@Override
	public String toString() {
		return "Account [id=" + id + ", email=" + email + ", userName=" + userName + ", firstName=" + firstName
				+ ", lastName=" + lastName + ", fullName=" + fullName + ", createDate=" + createDate + "]";
	}

//	department="
//			+ department + ",

}

File: ReactJS/HelloWorld/Account.ts
export class Account {
    private static counter: number = 0;  

    private accountId: number;
    private email: string;
    private username: String;
    private fullname: String;
    private departmentId: number;
    private positionId: number;
    private createDate: Date;


	constructor(email: string, username: String, fullname: String, departmentId: number, positionId: number, createDate: Date) {
		this.accountId = ++Account.counter;
		this.email = email;
		this.username = username;
		this.fullname = fullname;
		this.departmentId = departmentId;
		this.positionId = positionId;
		this.createDate = createDate;
	}

    /**
     * Getter accountId
     * @return {number}
     */
	public getAccountId(): number {
		return this.accountId;
	}

    /**
     * Getter email
     * @return {string}
     */
	public getEmail(): string {
		return this.email;
	}

    /**
     * Getter username
     * @return {String}
     */
	public getUsername(): String {
		return this.username;
	}

    /**
     * Getter fullname
     * @return {String}
     */
	public getFullname(): String {
		return this.fullname;
	}

    /**
     * Getter departmentId
     * @return {number}
     */
	public getDepartmentId(): number {
		return this.departmentId;
	}

    /**
     * Getter positionId
     * @return {number}
     */
	public getPositionId(): number {
		return this.positionId;
	}

    /**
     * Getter createDate
     * @return {Date}
     */
	public getCreateDate(): Date {
		return this.createDate;
	}

    /**
     * Setter accountId
     * @param {number} value
     */
	public setAccountId(value: number) {
		this.accountId = value;
	}

    /**
     * Setter email
     * @param {string} value
     */
	public setEmail(value: string) {
		this.email = value;
	}

    /**
     * Setter username
     * @param {String} value
     */
	public setUsername(value: String) {
		this.username = value;
	}

    /**
     * Setter fullname
     * @param {String} value
     */
	public setFullname(value: String) {
		this.fullname = value;
	}

    /**
     * Setter departmentId
     * @param {number} value
     */
	public setDepartmentId(value: number) {
		this.departmentId = value;
	}

    /**
     * Setter positionId
     * @param {number} value
     */
	public setPositionId(value: number) {
		this.positionId = value;
	}

    /**
     * Setter createDate
     * @param {Date} value
     */
	public setCreateDate(value: Date) {
		this.createDate = value;
	}


}